File: solutions/ruby/high-scores/1/high_scores.rb
class HighScores
  def initialize(arr)
    @scores = arr
  end

  def scores
    @scores
  end
  
  def latest
    @scores[-1]
  end

  def personal_best
    @scores.max
  end

  def personal_top_three
    @scores.sort_by{-_1}[0..2]
  end

  def latest_is_personal_best?
    personal_best == latest
  end
end

File: solutions/dart/high-scores/1/lib/high_scores.dart
import 'dart:math';
class HighScores {
  List<int> scores;
  HighScores(this.scores);

  int latest() {
    return scores.last;
  }

  int personalBest() {
    return scores.reduce((ths, nxt) => ths > nxt ? ths : nxt);
  }

  List<int> personalTopThree() {
    final srted = List<int>.from(scores)
      ..sort((a,b) => b.compareTo(a));
    return srted.sublist(0, min(srted.length,3));
  }
}
